src/NServiceBus.AzureStoragePersistence/CloudTableExtensions.cs

```diff
@@ -0,0 +1,26 @@
+﻿namespace NServiceBus
+{
+    using Microsoft.WindowsAzure.Storage.Table;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    public static class CloudTableExtensions
+    {
+        public static async Task<IList<T>> ExecuteQueryAsync<T>(this CloudTable table, TableQuery<T> query, CancellationToken ct = default(CancellationToken)) where T : ITableEntity, new()
+        {
+            var items = new List<T>();
+            var token = (TableContinuationToken)null;
```

Comment:
I seem to remember getting build failures when not using `var`. But now that you mention it, it could be only for item's that aren't assigned.